compliance_checker/ioos.py

```diff
@@ -346,6 +348,353 @@ def check_units(self, ds):
         cf16 = CF1_6Check()
         return cf16.check_units(ds)
 
+class IOOS1_2Check(IOOSNCCheck):
+    """
+    Class to implement the IOOS Metadata 1.2 Specification
+    """
+
+    _cc_spec_version = '1.2'
+    _cc_description = 'IOOS Metadata Profile, Version 1.2'
+    _cc_url = 'https://ioos.github.io/ioos-metadata/ioos-metadata-profile-v1-2.html'
+    register_checker = True
+
+    def __init__(self):
+
+        # instantiate objects used for delegation
+        self.acdd1_6 = ACDD1_3Check()
+        self.cf1_7   = CF1_7Check()
+
+        # extend standard_names set to include QARTOD standard_names
+        self._qartod_std_names = [
+            "aggregate_quality_flag",
+            "attenuated_signal_test_quality_flag",
+            "climatology_test_quality_flag",
+            "flat_line_test_quality_flag",
+            "gap_test_quality_flag",
+            "gross_range_test_quality_flag",
+            "location_test_quality_flag",
+            "multi_variate_test_quality_flag",
+            "neighbor_test_quality_flag",
+            "rate_of_change_test_quality_flag",
+            "spike_test_quality_flag",
+            "syntax_test_quality_flag"
+        ]
+        self.cf1_7._std_names._names.extend(self._qartod_std_names)
+
+        # check geophysical variables have the following attrs:
+        self.check_var_attrs = (
+            ("_FillValue", BaseCheck.MEDIUM),
+            ("missing_value", BaseCheck.MEDIUM),
+            #( "standard_name", BaseCheck.HIGH # already checked in CF1_7Check.check_standard_name()
+            ("standard_name_uri", BaseCheck.MEDIUM),
+            #( "units", BaseCheck.HIGH # already checked in CF1_7Check.check_units()
+            ("platform", BaseCheck.HIGH),
+            #( "wmo_platform_code", BaseCheck.HIGH # only "if applicable", see check_wmo_platform_code()
+        )
+
+        # Define the global attributes
+        self.required_atts = [
+            'Conventions',
+            'creator_country',
+            'creator_email',
```

Comment:
I will add validators to email, phone, postal code, Conventions string, etc, in a branch off of this one.